### PR TITLE
feat(website): migrates website to typescript

### PR DESCRIPTION
**Pull request by ixahmedxi**
This is the first PR of many that will update the lexical website and its codebase.

This PR introduces the following changes:

- Migrates the codebase to TypeScript.
- Removes unused markdown page example.

### CODE REVIEW

File: packages/lexical-website/src/components/HomepageFeatures/index.tsx [hunk 1]
```diff
@@ -6,6 +6,8 @@
  *
  */
 
+/* eslint-disable @typescript-eslint/no-var-requires */
+
 import Translate from '@docusaurus/Translate';
 import clsx from 'clsx';
 import React from 'react';
```
Comment (thegreatercurve) [inline packages/lexical-website/src/components/HomepageFeatures/index.tsx:13]:
```suggestion
import * as React from 'react';
```

File: packages/lexical-website/tsconfig.json [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{
+  "extends": "@tsconfig/docusaurus/tsconfig.json",
```
Comment (thegreatercurve) [inline packages/lexical-website/tsconfig.json:2]: I don't know if this has any specific compiler options that are required by docusaurus, but you can always just extends our base config:

```suggestion
  "extends": "../../tsconfig.json",
```

**Comment by thegreatercurve** [comment]
LGTM. Just the build errors that come up block accepting the PR. To replicate the errors, run `npm run build` or `npm run ci-check` locally.

File: packages/lexical-website/tsconfig.json (hunk 2 above)
Comment (ixahmedxi) [reply packages/lexical-website/tsconfig.json:2]: We do need these docusaurus compile options for typescript to know the types and the`@site` imports, I couldn't extend from two tsconfigs so hence why this is happening, I could however add the docusaurus typescript options to the base tsconfig but I don't know how you guys feel about that one. I could also copy paste the base config options into the website tsconfig.

**Comment by thegreatercurve** [comment]
Closing this for the moment due to staleness.

**Pull request closed without merging**